tests/random_test.py

```diff
@@ -394,6 +394,27 @@ def testGammaGradType(self):
     # Should not crash with a type error.
     api.vjp(f, a, b)
 
+  @parameterized.named_parameters(jtu.cases_from_list(
+      {"testcase_name": "_lam={}_{}".format(lam, dtype),
+       "lam": lam, "dtype": onp.dtype(dtype).name}
+      for lam in [0.5, 5, 50, 500]
+      for dtype in [onp.int32, onp.int64]))
+  def testPoisson(self, lam, dtype):
```

Comment:
I would also consider adding a moment-matching test. I've used it as a sanity check for the samplers to catch bugs.

In TFP we've also used KS-like tests but might be a little overkill for this PR.

Reply:
We have KS-like tests for continuous distributions, but Poisson is discrete so that's what `_CheckChiSquared` is verifying.

Moment tests do seem like an easy sanity check to add (even if possibly slightly redundant), so I'll add a couple of those.